src/ImageSharp/PixelFormats/PixelBlenders/PorterDuffFunctions.Generated.tt

```diff
@@ -27,44 +27,39 @@ namespace SixLabors.ImageSharp.PixelFormats.PixelBlenders
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static Vector4 <#=blender#>Src(Vector4 backdrop, Vector4 source, float opacity)
         {
-            opacity = opacity.Clamp(0, 1);
-            source.W *= opacity;          
+            source.W *= opacity.Clamp(0,1);

```

Comment:
Are these called by per-row operations or are they purely for per-pixel operations?

Reply:
If they are called from bulk we shouldn't be clamping per pixel.

Reply:
These are being clamped per pixel. that's why I wanted to remove them and replace them with DebugGuards.

But by using DebugGuards, several tests fail due to out of range values, which prevents the PR to pass the coverage tests,  etc.

So, for now, I preffer to go back to Clamp, and solve any potential out of range issues in future PRs

Reply:
I think it'll be best to fix clamping here while we are working on it.

In the internal method `PorterDuffFunctions.NormalSrc(Vector4 backdrop, Vector4 source, float opacity)` the opacity is clamped before being used as a multiplier.

As far as I can see that method is only called by.

`PorterDuffFunctions.NormalSrc<TPixel>(TPixel backdrop, TPixel source, float opacity)`

`NormalSrc.Blend(TPixel background, TPixel source, float amount)`

`NormalSrc.BlendFunction(Span<Vector4> destination, ReadOnlySpan<Vector4> background, ReadOnlySpan<Vector4> source, float amount)`

`NormalSrc.BlendFunction(Span<Vector4> destination, ReadOnlySpan<Vector4> background, ReadOnlySpan<Vector4> source, ReadOnlySpan<float> amount)`

The clamping should take place in the caller methods not the callee, same for the other operations. Only the last method taking the `ReadOnlySpan<float> amount` requires per-pixel clamping of the values but that seems like something we could do with `Vector<float>` in the caller in supported architectures.

Reply:
Thankfully all the changes required can be done in the T4 templates.

Reply:
I'm happy to make the changes if you like?

Reply:
Sure, go for it!, it's just lunch time here! 😄